src/components/AddActivityForm/AddActivityForm.js

```diff
@@ -12,35 +12,35 @@ import {
 } from './AddActivityFormStyle';
 
 export default function AddActivityForm({onSetActivities}) {
-  const [formInput, setFormInput] = useState({id: '', name: '', location: '', duration: '', type: '', infos: ''});
+  const [formData, setFormData] = useState({id: '', name: '', location: '', duration: '', type: '', infos: ''});
 
   //----- Data from Input -----
   const handleChange = event => {
-    setFormInput({...formInput, [event.target.name]: event.target.value});
+    setFormData({...formData, [event.target.name]: event.target.value});
   };
   //----- Submithandler (add all input as object to state and closes modal after submit) -----
   const handleSubmit = event => {
     event.preventDefault();
     const newInfos =
-      formInput.infos.indexOf('https://') !== 0
-        ? (formInput.infos = 'https://' + formInput.infos.trim().toLowerCase())
-        : formInput.infos.trim().toLowerCase();
+      formData.infos.indexOf('https://') !== 0
+        ? (formData.infos = 'https://' + formData.infos.trim().toLowerCase())
+        : formData.infos.trim().toLowerCase();
     onSetActivities(current => [
```

Comment:
I would suggest to only pass the new activity object here and do the state update in the `App.js`:
```js
onSetActivities({
   id: nanoid(),
   name: formData.name.trim(),
   location: formData.location.trim().replace(/[^a-z]/gi, ' '), 
   duration: formData.duration,
   type: formData.type,
   infos: newInfos,
   isFavorite: false,
})
```
In App.js:
```js
  function setActivityHandler(value) {
    setActivities(current => [...current, value]);
  }
```
The `current` functionality is something which is used by the set function of the state. This article from the react docs might help: https://beta.reactjs.org/learn/sharing-state-between-components

Reply:
Ok, I think I'm on the right way of understanding where what happens. Thank you!